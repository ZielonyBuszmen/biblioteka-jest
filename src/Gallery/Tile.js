import React, { Component } from 'react';
import './Tile.css';

class Tile extends Component {
  render() {
    const { src, likes, date, comments, onClick } = this.props;
    return (
      <div className='Tile' onClick={onClick}>
        <img className='Tile__image' src={src} alt='dynamic'/>
        <div className='Tile__body'>
          <button className='mr-1 Tile__body-like-button'>
            <span role='img' aria-label='likes'>👍 +2</span> {likes}
          </button>
          <button className='mr-1 Tile__body-comment-button'>
            <span role='img' aria-label='comments'>💬</span> {comments.length}
          </button>
          <small className='Tile__body-date'>{date}</small>
        </div>
      </div>
    );
  }
}

export default Tile;
