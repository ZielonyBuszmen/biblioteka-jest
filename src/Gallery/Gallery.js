import React, { Component } from 'react';
import Tile from './Tile';

class Gallery extends Component {

  onImageClick = (imageId) => {
    this.props.gotoImageHandler(imageId);
  };

  imageListRenderer = (imageList) => {
    return imageList.map(
      (image, id) => <Tile
        key={id}
        src={image.image}
        date={image.date}
        likes={image.likes}
        comments={image.comments}
        onClick={() => this.onImageClick(id)}
      />
    );
  };
  

  render() {
    return (
      <div className='row my-4'>
        <span>Obrazek:</span>
        {this.imageListRenderer(this.props.imageList)}
      </div>
    );
  }
}

export default Gallery;
