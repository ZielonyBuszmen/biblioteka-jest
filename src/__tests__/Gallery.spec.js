import renderer from 'react-test-renderer';
import React from 'react';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-enzyme';
import Gallery from '../Gallery/Gallery';

configure({adapter: new Adapter() });

describe("Gallery", () => {
    it("should render", () => {
        const imageList = [
            {
                image: 'photo.jpg',
                date: '20 July',
                likes: 13,
                comments: [],
            },
        ];

        const component = renderer
            .create(<Gallery imageList={imageList}/>)
            .toJSON();

        expect(component).toMatchSnapshot();
    });

    it("should render with enzyme", () => {
        const imageList = [
            {
                image: 'photo.jpg',
                date: '20 July',
                likes: 13,
                comments: [],
            },
        ];

        const component = shallow(<Gallery imageList={imageList} />);
        expect(component).toMatchSnapshot();

    });
});