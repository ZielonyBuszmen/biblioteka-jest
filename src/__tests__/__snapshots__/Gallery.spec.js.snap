// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery should render 1`] = `
<div
  className="row my-4"
>
  <span>
    Obrazek:
  </span>
  <div
    className="Tile"
    onClick={[Function]}
  >
    <img
      alt="dynamic"
      className="Tile__image"
      src="photo.jpg"
    />
    <div
      className="Tile__body"
    >
      <button
        className="mr-1 Tile__body-like-button"
      >
        <span
          aria-label="likes"
          role="img"
        >
          üëç +2
        </span>
         
        13
      </button>
      <button
        className="mr-1 Tile__body-comment-button"
      >
        <span
          aria-label="comments"
          role="img"
        >
          üí¨
        </span>
         
        0
      </button>
      <small
        className="Tile__body-date"
      >
        20 July
      </small>
    </div>
  </div>
</div>
`;

exports[`Gallery should render with enzyme 1`] = `
<div
  className="row my-4"
>
  <span>
    Obrazek:
  </span>
  <Tile
    comments={Array []}
    date="20 July"
    key="0"
    likes={13}
    onClick={[Function]}
    src="photo.jpg"
  />
</div>
`;
